### PR TITLE
Day 6 - React - Controlled Inputs

### PR DESCRIPTION
**Pull request by AnimeshKumarSuccessive**
Animesh Kumar
ticket number ; [40535](https://redmine.successive.tech/issues/40535)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
-import TextFieldDemo from './pages/TextFieldDemo/TextFieldDemo';
+import InputDemo from './pages/InputDemo';
 
 function App() {
   return (
     <div>
-      <TextFieldDemo />
+      {/* <TextFieldDemo /> */}
```
Comment (yogeshsinghh) [inline src/App.jsx]: remove commented code

File: src/components/RadioGroup/RadioGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const RadioGroup = (props) => {
+  const {
+    value,
```
Comment (yogeshsinghh) [inline src/components/RadioGroup/RadioGroup.jsx:6]: rename value as selectedValue instead from the options

File: src/components/RadioGroup/RadioGroup.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <>
+      <div value={value}>
+        { value
+          ? (
+            <span>
+              <b>
+                What you do?
+              </b>
+            </span>
+          ) : ''}
```
Comment (yogeshsinghh) [inline src/components/RadioGroup/RadioGroup.jsx]: move this block to inputDemo.js as this component should be generic

File: src/components/RadioGroup/RadioGroup.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        {options.map((item) => {
+          const { value: selectValue, label } = item;
+          if (selectValue !== value) return false;
```
Comment (yogeshsinghh) [inline src/components/RadioGroup/RadioGroup.jsx]: remove this condition and pass only those options on the basis of selected sport

File: src/components/SelectField/SelectField.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  const {
+    value, error, onChange, options, defaultText,
+  } = props;
+  return (
+    <div>
+      <div>
+        <b style={boldlabel}>
+          Select the game you want to play?
+        </b>
+      </div>
```
Comment (yogeshsinghh) [inline src/components/SelectField/SelectField.jsx:15]: move this block to inputDemo.js as this component should be generic

File: src/config/constant.js [hunk 6]
```diff
@@ -1,10 +1,42 @@
...
+
+];
+
+export const radioOptions = [
+  {
+    value: 'cricket',
+    label: 'Wicket-Keeper',
+  },
+  {
+    value: 'cricket',
```
Comment (yogeshsinghh) [inline src/config/constant.js]: radio options values should be unique

File: src/pages/InputDemo/InputDemo.jsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    });
+  };
+
+  const handleNameChange = (event) => {
+    setValue({
+      name: event.target.value,
+      sport: value.sport,
+      cricket: value.cricket,
+      football: value.football,
+    });
```
Comment (yogeshsinghh) [inline src/pages/InputDemo/InputDemo.jsx]: use spread operator

File: src/pages/InputDemo/InputDemo.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  const handleSportProfileChange = (event) => {
+    setValue({
+      name: value.name,
+      sport: value.sport,
+      cricket: value.sport && value.sport === 'cricket' ? event.target.attributes.label.value : '',
+      football: value.sport && value.sport === 'football' ? event.target.attributes.label.value : '',
```
Comment (yogeshsinghh) [inline src/pages/InputDemo/InputDemo.jsx]: set value from  the event.target.value

**Review by yogeshsinghh — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (AnimeshKumarSuccessive) [reply src/App.jsx]: Done

File: src/components/RadioGroup/RadioGroup.jsx (hunk 2 above)
Comment (AnimeshKumarSuccessive) [reply src/components/RadioGroup/RadioGroup.jsx:6]: Done

File: src/components/RadioGroup/RadioGroup.jsx (hunk 3 above)
Comment (AnimeshKumarSuccessive) [reply src/components/RadioGroup/RadioGroup.jsx]: Done

File: src/components/SelectField/SelectField.jsx (hunk 5 above)
Comment (AnimeshKumarSuccessive) [reply src/components/SelectField/SelectField.jsx:15]: Done

File: src/config/constant.js (hunk 6 above)
Comment (AnimeshKumarSuccessive) [reply src/config/constant.js]: Done

File: src/components/RadioGroup/RadioGroup.jsx (hunk 4 above)
Comment (AnimeshKumarSuccessive) [reply src/components/RadioGroup/RadioGroup.jsx]: Done

File: src/pages/InputDemo/InputDemo.jsx (hunk 8 above)
Comment (AnimeshKumarSuccessive) [reply src/pages/InputDemo/InputDemo.jsx]: done

**Review by yogeshsinghh — APPROVED** [review]

**Pull request merged**